AnomalyDetection/Core/src/main/java/org/tribuo/anomaly/AnomalyFactory.java

```diff
@@ -51,6 +54,32 @@ public final class AnomalyFactory implements OutputFactory<Event> {
 
     private static final AnomalyEvaluator evaluator = new AnomalyEvaluator();
 
+    /**
+     * Create an AnomalyFactory.
+     */
+    public AnomalyFactory() {}
+
+    /**
+     * Deserialization factory.
+     * @param version The serialized object version.
+     * @param className The class name.
+     * @param message The serialized data.
+     */
+    public static AnomalyFactory deserializeFromProto(int version, String className, Any message) {
+        if (version < 0 || version > 0) {
+            throw new IllegalArgumentException("Unknown version " + version + ", this class supports at most version " + 0);
+        }
+        if (message.getValue() != ByteString.EMPTY) {
+            throw new IllegalArgumentException("Invalid proto");
+        }
+        return new AnomalyFactory();
+    }
+
+    @Override
+    public OutputFactoryProto serialize() {
```

Comment:
you annotated this class with:

`@ProtoSerializableClass(version = 0)`

Then you could use the default serialization:

```
    @Override
    public OutputFactoryProto serialize() {
        return ProtoUtil.serialize(this);
    }

```